### DISCUSSION TITLE
Using GoRouter with Drawer in AppBar

### DISCUSSION THREAD

**Discussion by joshmikow** [csells/go_router#339] [Q&A]
Hello! I'm new to Flutter and trying to use GoRouter for the navigation within my App. Sorry if this is a simple question but I couldn't find any examples that helped me figure it out. 

Everything was working well with some nested routes until I tried to implement a Drawer within the AppBar for my users to get to settings and other screens. Now the drawer takes over the back button that is created once you navigate down to a nested router. 

I found an example of someone using a Row to add both a back button and a drawer next to each other but I'm not clear on how I would implement that with GoRouter handling the back button functionality. 

Any tips on how to make them work together?

**Comment by csells** [comment]
Actually, that's a good question; I've played around with this myself and gotten unsatisfying results. @esDotDev does this kind of stuff all the time. Maybe he'll have a suggestion.

**Comment by esDotDev** [comment]
Can you post a video showing more clearly your issue? I don't really understand.
